pkg/controller/constrainttemplate/constrainttemplate_controller_test.go

```diff
@@ -164,46 +181,29 @@ violation[{"msg": "denied!"}] {
 
 	// events will be used to receive events from dynamic watches registered
 	events := make(chan event.GenericEvent, 1024)
-	rec, _ := newReconciler(mgr, opaClient, wm, cs, tracker, events, events, func(context.Context) (*corev1.Pod, error) { return pod, nil })
+	rec, err := newReconciler(mgr, opaClient, wm, cs, tracker, events, events, func(context.Context) (*corev1.Pod, error) { return pod, nil })
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	err = add(mgr, rec)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	cstr := newDenyAllCstr()
-
 	ctx := context.Background()
 	testutils.StartManager(ctx, t, mgr)
 
-	// Clean up to remove the crd, constraint and constraint template
-	t.Cleanup(func() {
-		crd := &apiextensionsv1.CustomResourceDefinition{}
-		err := c.Get(ctx, crdKey, crd)
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		if err := deleteObjectAndConfirm(ctx, c, cstr, timeout); err != nil {
-			t.Fatalf("want deleteObjectAndConfirm(ctx, c, cstr, timeout) error = nil, got %v", err)
-		}
-		if err := deleteObjectAndConfirm(ctx, c, crd, timeout); err != nil {
-			t.Fatalf("want deleteObjectAndConfirm(ctx, c, crd, timeout) error = nil, got %v", err)
-		}
-		if err := deleteObjectAndConfirm(ctx, c, instance, timeout); err != nil {
-			t.Fatalf("want deleteObjectAndConfirm(ctx, c, instance, timeout) error = nil, got %v", err)
-		}
-	})
-
-	logger.Info("Running test: CRD Gets Created")
 	t.Run("CRD Gets Created", func(t *testing.T) {
-		err = c.Create(ctx, instance)
-		if err != nil {
-			t.Fatal(err)
-		}
+		logger.Info("Running test: CRD Gets Created")
+		constraintTemplate := makeReconcileConstrainTemplate()
+		t.Cleanup(deleteObjectAndConfirm(ctx, t, c, crdToDelete))
```

Comment:
deepcopy crdToDelete?

Reply:
No, DeepCopy here is unnecessary. I've added explanatory comments to "deleteObjectAndConfirm".